### ISSUE TITLE
Skip ad pop-up not showing?

### ISSUE THREAD

**Issue by igonro** [x0a/uBO-YouTube#10]
Hi, I love this extension; but when an ad is played, the "Skip ad" pop-up (that usually is available after 3-5 seconds) doesn't show up. I know that I can "Force skip ad", but if I click that option, does it count for the youtuber?

I want that some youtubers could earn money for his amazing content, but it's a little hard to follow up certain video if the ad is 2 minutes long.

Thanks for your amazing work!

**Comment by igonro** [comment]
Also I should ad that I can't see when the ads will be displayed (by default YouTube shows a yellow marks to indicate that an ad will be displayed at certain timestamp). It would be nice to have an option to disable/enable this.

**Comment by x0a** [comment]
Hey, 
Thanks for the report and happy to hear you like the extension! We were dealing with this issue not that long ago in #8 and pushed a fix, but was having a bit of trouble getting the updates to push to the Chrome/FF store. Could you go to https://youtube.com/ubo-yt#misc and tell me what version you are on? and what browser/version you're using? Should help let me know if this is a distribution problem or something new.

And to answer your question, if you press the 'Skip Ad' button, generally the channel doesn't get paid. But depending on the advertisers preference, they might decide to pay if you watched at least 30 seconds of the ad before skipping. My understanding is that unskippable ads always result in payout. In any case, 'Force Skip' is actually more likely to result in ad revenue than using YouTube's 'Skip Ad' button because it works by seeking to the end of the ad. Sometimes this sends a XHR ping letting the server know you "finished" watching the ad. From my own (limited) research, this results in revenue about 15-30% of the time, which is higher than using the skip-ad button. But I still recommend using it only sparingly.

As far as the yellow ad indicators being removed, unfortunately it seems that this was an intentional change by YouTube: https://support.google.com/youtube/thread/9664157?hl=en https://www.reddit.com/r/youtube/comments/cg16tw/ads_are_now_hidden_why_is_this_a_thing/

No guarantees but I might look into re-introducing these as an optional feature if I can find a predictable source for the timestamps.

**Comment by igonro** [comment]
Hi,

I'm using Firefox 72.0.1 and the extension version is 0.9.7. I don't know if the problem is with this extension, because sometimes the "Skip ad" button it appears, and sometimes it doesn't.

Ok. Now that I know how the "force skip" button works I will use it! I usually try to watch the relevant ads, but skip the boring/not relevant ones.

Oh, don't worry with the yellow marks then. It's okay this way!

Thanks for your response and your amazing work!

**Comment by x0a** [comment]
Hey,

As it turns out there was a distribution problem. The version with the fix was 0.9.8 but the Mozilla team hadn't approved the update. I submitted a new version and it should be available for update now. Please update your extensions (Menu -> Add-ons -> Extensions -> Click on the gear -> Check for updates) and feel free to re-open this issue if you still experience the issue.

Thanks
